### ISSUE TITLE
Plugins / Atomic: Hide additional update notification indicators for symlinked plugins

### ISSUE THREAD

**Issue by JoshuaGoode** [Automattic/wp-calypso#58590]
### Quick summary

On WordPress.com on Atomic, we hide update notifications for plugins we symlink and manage.  This is done via wpcomsh.php

This sometimes is a bit more overbearing than necessary. Re: https://github.com/Automattic/wp-calypso/issues/52511

However, the current attempts to hide update notifications are not always enough.

#### Example

In some cases, a symlinked plugin will have an update available which we will intentionally not apply yet.

When this occurs, the site and updater will believe an update is available and we prevent it from updating and attempt to hide the update notices.

When a site is in this state, we can visit My Site > My Home > Updates, `/wp-admin/update-core.php`.  There, we will see a small update icon in the sidebar but no update will be available.

![Screen Shot on 2021-11-29 at 10:34:17](https://user-images.githubusercontent.com/4887830/143897487-1b0a949d-8158-4264-ba7a-3632272e1418.png)

We can also visit Calypso's My Site > Plugins page at `https://wordpress.com/plugins/SITE` and see a similar notification.

![Screen Shot on 2021-11-29 at 10:35:00](https://user-images.githubusercontent.com/4887830/143897645-0019dabc-b534-445f-b63c-76e028bf32ff.png)

### Steps to reproduce

First, a site must have a symlinked plugin that the updater API detects that there's an update for on the .org repo.

Internal notes on symlinks: P9HQHe-Ge-p2

Automatticians can easily check this by visiting accessing the site's WP-CLI and running `plugin list`.   Then, check for symlinked plugins with updates available.

If an update is available for a symlinked plugin, we can then visit the following.

1. My Site > My Home > Updates, `/wp-admin/update-core.php` -- Observe that there is an update notification on the sidebar next to `Updates` but no actionable update available.

2. Visit Calypso's My Site > Plugins page (not the WP-Admin), `https://wordpress.com/plugins/SITE` --  Observe an update notification in the sidebar next to `Plugins`.

### What you expected to happen

No update notifications of any kind to be visible for symlinked plugins. 

### What actually happened

Update notifications appear but are not actionable. 

### Context

Internal request: p9F6qB-7Kq-p2

4541234-zd-woothemes

Somewhat related issue:

https://github.com/Automattic/wp-calypso/issues/52511

### Simple, Atomic or both?

Atomic

**Comment by mmtr** [comment]
@sejas and I have been taking a look at this and prepare a PR to fix this issue (945-gh-Automattic/wpcomsh). However, today we noticed that we might don't need it at all since the issue could be already fixed?

Gutenberg released yesterday a new version (13.2.0) but Atomic sites are still using the previous version (13.1.0). On our tests, the "My Home > Updates" and "Plugins" menus show the expected numbers (1 for "My Home > Updates" since there is a new translation, and nothing for "Plugins"):

<img width="834" alt="Screen Shot 2022-05-12 at 13 00 36" src="https://user-images.githubusercontent.com/1233880/168056764-e53e35a5-85ee-42dd-9bc1-253714911161.png">

<img width="1066" alt="Screen Shot 2022-05-12 at 13 01 13" src="https://user-images.githubusercontent.com/1233880/168056783-f797cb86-1418-477b-9584-3acc12efad0d.png">

<img width="1667" alt="Screen Shot 2022-05-12 at 13 01 28" src="https://user-images.githubusercontent.com/1233880/168056802-4f3f6793-613b-4368-95ab-989054aa172e.png">

@JoshuaGoode are you able to still reproduce this issue or should we close it?

**Comment by JoshuaGoode** [comment]
Thanks for digging in.

It appears this is no longer an issue @mmtr  @sejas -- I can no longer reproduce my original report.

**Comment by sejas** [comment]
Thank you for checking again, @JoshuaGoode !
I'm closing this issue.

**Comment by kriskorn** [comment]
@sejas I am going to open this up again, as it seems this notification is again showing when it should not. Here is a quick example of how this shows up at the moment on a customer site. More information can be found in p1665616053382419-slack-wpcom-support

**Calypso**
<img width="712" alt="calypso-plugin-notification" src="https://user-images.githubusercontent.com/12064241/195809085-e6e2fdf2-9aaf-4f32-8805-a6946d60f6e1.png">

**WP Admin**
<img width="891" alt="wp-admin-plugin-notification" src="https://user-images.githubusercontent.com/12064241/195809155-0b83e596-0420-43aa-99f9-81eb01e32f84.png">

**CLI**

```
$ wp plugin list
+---------------------------------------------------+----------+------------------------------+----------+
| name                                              | status   | update                       | version  |
+---------------------------------------------------+----------+------------------------------+----------+
| akismet                                           | active   | none                         | 5.0.1    |
| amp                                               | inactive | none                         | 2.3.0    |
| badges-woo                                        | active   | none                         | 1.1.0    |
| checkout-field-editor-and-manager-for-woocommerce | active   | none                         | 2.2.2    |
| classic-editor                                    | active   | none                         | 1.6.2    |
| crowdsignal-forms                                 | active   | none                         | 1.6.6    |
| currencyconverter                                 | active   | none                         | 0.5.4    |
| woocommerce-direct-checkout                       | active   | none                         | 2.7.0    |
| duplicate-page                                    | active   | none                         | 4.4.9    |
| facebook-messenger-customer-chat                  | active   | none                         | 2.5      |
| facebook-for-woocommerce                          | active   | available                    | 2.6.25   |
| fluid-checkout                                    | active   | none                         | 2.0.7    |
| fullwidth-templates                               | active   | none                         | 1.1.1    |
| google-listings-and-ads                           | active   | none                         | 2.1.4    |
| gutenberg                                         | active   | none                         | 14.3.0   |
| header-footer-code-manager                        | active   | none                         | 1.1.29   |
| health-check                                      | active   | none                         | 1.5.0    |
```

As we can see **Facebook for WooCommerce** has an update available, but it is a managed plugin and that should not show a notification. And it seems it is working correctly in WP Admin ( no notification shown ), but not in Calypso. 

There are also 3 updates available for 3 different themes, but they are also all managed by us, so no notifications should not show for them either.

**Comment by happychait** [comment]
This is happening again as reported persistently in this user report: 39159030-hc

**Comment by danielbachhuber** [comment]
@happychait Thanks! It seems like there might still be something wonky. Some additional Slack conversation in p1673285064874299-slack-C7YPW6K40

**Comment by sejas** [comment]
@happychait , After debugging this particular site, I've found that the update badge comes from [`fonts-plugin-pro`](https://fontsplugin.com/). I'm not sure why the Plugin update notice doesn't appear in `update-core.php` or the plugins list. It doesn't appear in `wp plugin list` either. However, the same behavior is present in Core.

I recommend suggesting the user to update the `Fonts Plugin Pro` plugin manually to the latest version (1.7.0).

- Current plugin version: `1.4.7` <img width="671" alt="Screenshot 2023-01-10 at 05 48 17" src="https://user-images.githubusercontent.com/779993/211471548-e4776508-7202-4d48-95de-60037d982ee2.png">
- Version available: `1.7.0`

Currently, the best way to troubleshoot this "issue" is running: `wp transient get update_plugins --network` and determine if the update badge comes from a third-party plugin.:
```
array (
  'fonts-plugin-pro/inc/updates/update.php' => 
  (object) array(
     'new_version' => '1.7.0',
     'stable_version' => '1.7.0',
     'name' => 'Fonts Plugin Pro',
     'slug' => 'update',
     'url' => 'https://fontsplugin.com/downloads/fonts-plugin-pro/?changelog=1',
     'last_updated' => '2022-11-24 14:27:50',
     'homepage' => 'https://fontsplugin.com/',
...
     'plugin' => 'fonts-plugin-pro/inc/updates/update.php',
     'id' => 'fonts-plugin-pro/inc/updates/update.php',
  ),
)
```

Feel free to let me know if I can help with anything else.

**Comment by danielbachhuber** [comment]
@sejas Great job debugging! I appreciate your persistence 

@ the reporters: We're going to assume this issue is fixed now. Feel free to re-open this issue or create a new issue if there are additional reports.
